src/SFML/Graphics/Image.cpp

```diff
@@ -28,85 +28,75 @@
 #include <SFML/Graphics/Image.hpp>
 #include <SFML/Graphics/ImageLoader.hpp>
 #include <SFML/System/Err.hpp>
+
+#include "SFML/System/Vector2.hpp"
 #ifdef SFML_SYSTEM_ANDROID
 #include <SFML/System/Android/ResourceStream.hpp>
 #endif
 #include <algorithm>
 #include <cstring>
 #include <ostream>
+#include <utility>
 
 
 namespace sf
 {
 ////////////////////////////////////////////////////////////
-void Image::create(const Vector2u& size, const Color& color)
+Image::Image(Vector2u size, std::vector<std::uint8_t>&& pixels) : m_size(size), m_pixels(std::move(pixels))
```

Comment:
Question, do these two parameters have the same runtime semantics in terms of optimization/copies?
```cpp
std::vector<std::uint8_t> pixels
std::vector<std::uint8_t>&& pixels
```
Even for different argument types (temporary or not, const or not, ...)

Reply:
They do not -- when passing an rvalue, the first one will result in two moves, the second one only in one move.